scipy/stats/_hypotests.py

```diff
@@ -836,52 +836,59 @@ def barnard_exact(table, alternative="two-sided", pooled=True, num_it=3):
     should only be used when both sets of marginals are fixed.
 
     """
-    if num_it <= 0:
+    if n_iter <= 0:
         raise ValueError(
             "Number of iterations `num_it` must be strictly positive, "
-            f"found {num_it!r}"
+            f"found {n_iter!r}"
         )
 
-    c = np.asarray(table, dtype=np.int64)
+    table_nparray = np.asarray(table, dtype=np.int64)
```

Comment:
Why not just `table`? `_nparray` does not bring more info here.

Reply:
> Why not just `table`? `_nparray` does not bring more info here.

Yes, you are right, change it in : 057c95526adef024ff4d5c50fa10c89df55791cc